server/src/formulas/codeSnippets.ts

```diff
@@ -0,0 +1,27 @@
+import { Formula, FormulaPreview } from '../types/formulas';
+
+export const formulas: Array<FormulaPreview> = [
+  {
+    id: Formula.WeightedAverage,
+    name: 'Weighted average',
+    attributes: ['maxPoints', 'minRequiredPoints', 'weight'],
+    codeSnippet:
+    `const weightedAverage = (nums, weights) => {
+      const [sum, weightSum] = weights.reduce(
+        (acc, w, i) => {
+          acc[0] = acc[0] + nums[i] * w;
+          acc[1] = acc[1] + w;
+          return acc;
+        },
+        [0, 0]
+      );
+      return sum / weightSum;
+    };`
+  },
+  {
+    id: Formula.Manual,
+    name: 'Manual',
+    attributes: ['maxPoints', 'minRequiredPoints'],
+    codeSnippet: 'placeholder'
+  }
+];
```

Comment:
Rather than having these all here, we could also store this preview information in `formulas/index.ts` and include per-formula information in the same `registerFormula` call in each formula definition file.

Reply:
moved the preview info of each formula to respective formula file, these are now registered to the `formulaImplementations`